stdlib/public/SwiftShims/RefCount.h

```diff
@@ -185,15 +185,16 @@ _swift_release_dealloc(swift::HeapObject *object)
 
 namespace swift {
 
-// FIXME: many `relaxed` in this file should be `consume`,
-// but (1) the compiler doesn't support `consume` directly,
-// and (2) the compiler promotes `consume` to `acquire` instead which
-// is overkill on our CPUs. But this might leave us vulnerable to
-// compiler optimizations that `relaxed` allows but `consume` ought not.
-#define relaxed std::memory_order_relaxed
-#define acquire std::memory_order_acquire
-#define release std::memory_order_release
-#define consume std::memory_order_consume
+// FIXME: some operations here should be memory_order_consume, 
+// but (1) the compiler doesn't support consume directly,
+// and (2) the compiler implements consume as acquire which
+// is unnecessarily slow on some of our CPUs.
+// Such operations are written here as fake_memory_order_consume.
+// We map them to memory_order_relaxed. This might leave us vulnerable to
+// compiler optimizations. In addition, the other dependency-tracking
+// annotations that would be required for real memory_order_consume
+// are not present.
+#define fake_memory_order_consume std::memory_order_relaxed
```

Comment:
I haven't looked for codegen differences on Intel here. You're right, if there are differences on Intel then they are likely to be signs of optimizations that could break fake-consume.